Modules/Forum/classes/class.ilObjForumGUI.php

```diff
@@ -4457,6 +4826,7 @@ public function mergeThreadsObject(): void
         }
 
         $threadIdToMerge = (int) ($this->httpRequest->getQueryParams()['merge_thread_id'] ?? 0);
+        $threadIdToMerge = (int) ($this->http->wrapper()->query()->retrieve('thr_pk', $this->refinery->kindlyTo()->int()) ?? 0);
```

Comment:
This makes not sense. The transformation either returns an integer, or raises an exception. NULL is not possible.

Please try this.

```php
$threadIdToMerge = $this->http->wrapper()->query()->retrieve(
    'thr_pk',
    $this->refinery->byTrying([
        $this->refinery->kindlyTo()->int(),
        $this->refinery->always(0)
    ])
);
```